xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,165 @@
+def main(args: argparse.Namespace) -> None:
+    try:
+        with open(json_file_path, 'r') as json_file:
+            data = json.load(json_file)
+
+        for attribute_name in data.keys():
+            print(attribute_name)
+            attribute_data = data[attribute_name]
+
+            # Ensure that the test output directory is created
+            output_dir_path = attribute_data["path"] + "-output/"
+            if not os.path.exists(output_dir_path):
+                try:
+                    os.makedirs(output_dir_path)
+                except OSError as e:
+                    print(f"Error: {e}")
+
+            for test in attribute_data["tests"].keys():
+                file_name: str = attribute_name + test + ".xml"
+                input_xml_path = os.path.join(attribute_data["path"], file_name)
+                output_xml_path = os.path.join(output_dir_path, file_name)
+
+                result = run_xml_converter(input_xml=[input_xml_path], output_xml=[output_xml_path])
+                # Remove noisy lines
+                stdout = remove_noisy_lines(result.stdout.split("\n"))
+                stderr = remove_noisy_lines(result.stderr.split("\n"))
+                xml_diff = compare_files(input_xml_path, output_xml_path)
+
+                # Prints the results rather than comparing them to a file
+                if args.print:
+                    print(f"Test {attribute_name}{test}")
+                    print(f"'output_stdout' : {json.dumps(stdout)}")
+                    print(f"'output_stderr' = {json.dumps(stderr)}")
+                    print(f"'expected_xml_diff' = {json.dumps(xml_diff)}")
+                    continue
+
+                all_tests_passed: bool = True
+                expected_results = attribute_data["tests"][test]
+
+                if stdout != expected_results["output_stdout"]:
+                    print(f"Output did not match for test {attribute_name}{test}")
+                    print(f"Expected stdout {expected_results['output_stdout']} and got {stdout}")
+                    all_tests_passed = False
+
+                if stderr != expected_results["output_stderr"]:
+                    print(f"Output did not match for test {attribute_name}{test}")
+                    print(f"Expected stderr {expected_results['output_stderr']} and got {stderr}")
+                    all_tests_passed = False
+
+                if xml_diff != expected_results["expected_xml_diff"]:
+                    print(f"Diff was incorrect for test {attribute_name}{test}")
+                    print(f"Expected {expected_results['expected_xml_diff']} and got {xml_diff}")
+                    all_tests_passed = False
+
+                if all_tests_passed:
+                    print(f"Success: test {attribute_name}{test}")
+
+    except FileNotFoundError:
+        print(f"The file '{json_file_path}' does not exist.")
+    except json.JSONDecodeError as e:
+        print(f"JSON decoding error: {e}")
+    except Exception as e:
```

Comment:
This single line catches every possible error that could possibly occur anywhere inside this entire executable except for `FileNotFoundError`s, `JSONDecodeError`s, and the three lines of code that are outside of the `main()` function. Lets not do that.